src/pybind/mgr/zabbix/module.py

```diff
@@ -132,6 +137,20 @@ def set_config_option(self, option, value):
         self.config[option] = value
         return True
 
+    def _parse_zabbix_hosts(self):
+        self._zabbix_hosts = list()
+        servers = self.config['zabbix_host'].split(",")
+        for server in servers:
+            uri = re.match("(?:(?:\[?)([a-z0-9-\.]+|[a-f0-9:\.]+)(?:\]?))(?:((?::))([0-9]{1,5}))?$", server)
```

Comment:
I have not tested this one. But how does it handle IPv6 in combination with ports?

Reply:
Yes, you can test the regex here: https://regex101.com/r/x0yGnS/2

And here is a snippet from ceph-mgr log when I tried to provide your setting to `zabbix_host`:
```
2019-02-28 10:01:06.803038 7f8d154b8700  0 log_channel(audit) log [DBG] : from='client.514898 10.0.0.101:0/1663261630' entity='client.admin' cmd=[{"prefix": "zabbix config-set", "key": "zabbix_host", "value": "
[2001:db8::100],[2001:db8::200]:3333", "target": ["mgr", ""]}]: dispatch
2019-02-28 10:01:06.901871 7f8d15cb9700  0 mgr[zabbix] Parsed Zabbix hosts: [{'zabbix_host': '2001:db8::100', 'zabbix_port': 10051}, {'zabbix_host': '2001:db8::200', 'zabbix_port': '3333'}]
2019-02-28 10:01:13.430222 7f8d124b2700  4 mgr[zabbix] Sending data to Zabbix server 2001:db8::100, port 10051 as host/identifier Cloud
2019-02-28 10:01:13.442982 7f8d124b2700  4 mgr[zabbix] Sending data to Zabbix server 2001:db8::200, port 3333 as host/identifier Cloud
```

Reply:
Is there a reason not to use `urllib.parse`?

Reply:
It has no validation capability. It only parses URL string into specific fields. So I would need to validate its output anyway.

Reply:
Argh. Python needs a proper URL class in the standard library.